Source/Dafny/DafnyOptions.cs

```diff
@@ -582,7 +583,9 @@ public enum IncludesModes {
         XmlSink = new Bpl.XmlSink(this, BoogieXmlFilename);
       }
 
-      Compiler ??= new CsharpCompiler();
+      if (Compiler == null || Compiler is PlaceholderCompiler) {
```

Comment:
Compiler will never be null, right? Since you initialize it in the constructor. However, I would keep it that way in case someone else sets this field to null manually just to get the default.